internal/envoy/route.go

```diff
@@ -194,6 +193,40 @@ func UpgradeHTTPS() *envoy_api_v2_route.Route_Redirect {
 	}
 }
 
+func headerValueList(hvm map[string]string) (hvs []*envoy_api_v2_core.HeaderValueOption) {
+	for key, value := range hvm {
+		hvs = append(hvs, &envoy_api_v2_core.HeaderValueOption{
```

Comment:
If the user specified the "Host" header, should we generate a `host_rewrite`, or error out?

Reply:
If we want to do a separate Host rewrite API, then we should disallow Host here. My preference would be to accept the Host header here and do the right thing, but then I use curl a lot :)

Reply:
I already added logic (and tests) to exclude it in the validation logic (so a Host header should never reach here), but I can add something here too.  If we want to support host-rewriting through this API, then I'll have to dig some more about how to encoded that for Envoy, but was biasing away from scope creeping this change (and focusing on the minimum correct subset for this increment).

Reply:
Keeping the change subset here small is good, if we are going to release this in 1.1, it's already a substantial scope increase, and we'll need to be careful about checking it and @stevesloka's change for external services for consistency.

Reply:
Agree on rationalizing this with @stevesloka's change, I'd like to start looking into this more and left a few comments on it.  I've added a brief design doc (following the template) to this PR (I can split it off, not sure of the process?) that tries to capture the union of the two changes.  I believe the API in this change should be what's necessary and sufficient to implement both, but leaves the Host-rewrite mechanics for @stevesloka's PR.

Reply:
Ok, so @stevesloka merged the PR with the API design discussed yesterday, which should now be reflected here.  It's listed as "Draft", and I'm not sure how the lifecycle of these things progresses? 🤓

Reply:
I think we're going to need to land this PR before mine. They are essentially the same, with my PR having a lot less logic. 

However, my PR does handle host rewrites as a special case, in that if you specify `Host` as the key, we set different configuration on the Envoy side.